src/main/java/org/fcrepo/auth/webac/WebACAuthorizationDelegate.java

```diff
@@ -37,10 +49,45 @@
      */
     private static final Logger LOGGER = LoggerFactory.getLogger(WebACAuthorizationDelegate.class);
 
+    private static final Map<String, URI> actionMap;
+
+    static {
+        Map<String, URI> map = new HashMap<>();
+        map.put("GET", WEBAC_MODE_READ);
+        map.put("POST", WEBAC_MODE_APPEND);
+        map.put("PUT", WEBAC_MODE_WRITE);
+        map.put("DELETE", WEBAC_MODE_WRITE);
+        map.put("PATCH", WEBAC_MODE_WRITE);
+        map.put("OPTIONS", WEBAC_MODE_READ);
+        actionMap = Collections.unmodifiableMap(map);
```

Comment:
Incidentally, if @acoburn is correct, this is one reason why we don't want to have resources created or updated or deleted by requests to URIs that aren't their own.